wa/framework/workload.py

```diff
@@ -142,9 +143,57 @@ class ApkUiautoWorkload(ApkUIWorkload):
 
     platform = 'android'
 
+    parameters = [
+        Parameter('package', kind=str,
+                  description="""
+                  The pacakge name that can be used to specify
+                  the workload apk to use.
+                  """),
+        Parameter('install_timeout', kind=int,
+                  constraint=lambda x: x > 0,
+                  default=300,
+                  description="""
+                  Timeout for the installation of the apk.
+                  """),
+        Parameter('version', kind=str,
+                  default=None,
+                  description="""
+                  The version of the package to be used.
+                  """),
+        Parameter('variant', kind=str,
+                  default=None,
+                  description="""
+                  The variant of the package to be used.
+                  """),
+        Parameter('strict', kind=bool,
+                  default=False,
+                  description="""
+                  Whether to throw and error if the specified package cannot be found
+                  on host.
+                  """),
+        Parameter('force_install', kind=bool,
+                  default=False,
+                  description="""
+                  Always re-install the APK, even if matching version is found already installed
+                  on the device.
+                  """),
+        Parameter('uninstall', kind=bool,
+                  default=False,
+                  description="""
+                  If ``True``, will uninstall workload\'s APK as part of teardown.'
+                  """),
+    ]
+
     def __init__(self, target, **kwargs):
         super(ApkUiautoWorkload, self).__init__(target, **kwargs)
-        self.apk = PackageHandler(self)
+        self.apk = PackageHandler(self,
+                                  package=kwargs.get('package'),
```

Comment:
these are better done as  `package=self.package` and similar, as the super's `__init__` will populate the parameters from kwargs into instance members, and handle the setting of defaults, etc.